Sources/SwiftDocC/Indexing/Navigator/NavigatorIndex.swift

```diff
@@ -707,6 +707,22 @@ extension NavigatorIndex {
                 multiCuratedUnvisited.remove(normalizedIdentifier)
             }
             
+            // Check if the render node has a variant for Objective-C
+            let objCVariantTrait = renderNode.variantOverrides?.values.flatMap({ $0.traits }).first { trait in
+                switch trait {
+                case .interfaceLanguage(let language):
+                    return InterfaceLanguage.from(string: language) == .objc
+                }
+            }
+            
+            // In case we have a variant for Objective-C, apply the variant and re-index the render node.
+            if let variantToApply = objCVariantTrait {
+                let encodedRenderNode = try renderNode.encodeToJSON()
+                let transformedData = try RenderNodeVariantOverridesApplier().applyVariantOverrides(in: encodedRenderNode, for: [variantToApply])
```

Comment:
Is it worth just keeping around the JSON for every render node so you don't have to re-encode? Or is the perf cost not high enough to warrant the increase in peak memory usage?

Reply:
The indexing during conversion happens on in-memory `RenderNode` models, so we're not "re-encoding". The `RenderNodeVariantOverridesApplier` works on the JSON data rather than a Codable model, so there isn't really a better way to write this at the moment.

Reply:
Ahh I see, ok :thumbsup: